lib/auth/index.js

```diff
@@ -24,7 +24,7 @@ module.exports = settings => {
   const permissions = can(settings.permissions);
 
   router.use((req, res, next) => {
-    if (!req.session && !settings.bearerOnly) {
+    if (req.path !== '/logout' && !req.session && !settings.bearerOnly) {
       return next(new Error('No session'));
```

Comment:
Does this error display to the user? Would it be nicer UX to send them directly to the login if they've got no session?

Reply:
No, it's more a developer done f***ed up error that throws here before it throws a less scrutable error downstream. 

If they hit this there's no way for _the user_ to rectify it.

Reply:
The other thing that's always worth remembering is that throwing an error in code does not _necessarily_ mean that the user sees that error thrown (and often shouldn't), but it's generally easier and more scalable to handle errors in the error handler than to try and handle errors inline as entire classes of errors can be handled in a consistent way.

Reply:
Yeah I agree, throw a specific error type and handle them all in one place.